Framework/CurveFitting/src/CostFunctions/CostFuncUnweightedLeastSquares.cpp

```diff
@@ -53,6 +53,15 @@ std::vector<double> CostFuncUnweightedLeastSquares::getFitWeights(API::FunctionV
   return weights;
 }
 
+/// Update the fit weights of m_values so that correct fit weights are used by the minimizer and GSLFunctions
+void CostFuncUnweightedLeastSquares::updateFitWeights() {
+  for (size_t i = 0; i < m_values->size(); i++) {
+    if (m_values->getFitWeight(i) != 0.0) {
```

Comment:
Wat if the error does equal zero (i.e. empty bin) but we still want to include this point in the fit?
Would it also not be possible to set these weights to 1 in the first place (though I understand you're having to work around GSL here).